astropy/visualization/wcsaxes/tests/test_misc.py

```diff
@@ -21,9 +21,7 @@
 from astropy.visualization.wcsaxes.transforms import CurvedTransform
 
 mpl_version = Version(matplotlib.__version__)
-MATPLOTLIB_LT_21 = mpl_version < Version("2.1")
-MATPLOTLIB_LT_22 = mpl_version < Version("2.2")
-MATPLOTLIB_33 = mpl_version.major == 3 or mpl_version.minor == 3
+MATPLOTLIB_EQ_33 = mpl_version.major == 3 and mpl_version.minor == 3
```

Comment:
I think the `or` is a bug, is it, @astrofrog or @Cadair ? If it is a bug that needs backporting, I can open a separate PR, please let me know.

Reply:
Might refer to a (hypothetical) mpl 2.3 where the associated issue would be fixed as well as in all 3.x releases, but that's obviously not a very robust test.
I think it should actually be `MATPLOTLIB_GE_30 = mpl_version < Version("3.0")` (or rather be removed now); see below.

Reply:
it's a bug

Reply:
Pretty sure @Cadair meant for it to be 'and'

Reply:
Yes

Reply:
Does this bug need backporting to `<4.2`?

Reply:
if yes, that has to be a separate PR against the 4.0.x branch. Thanks!

Reply:
@bsipocz , what about 4.1.x branch?

Reply:
🤷 if there will a bugfix on that, then @eteq can cherry-pick it from 4.0.x? 
or if you want to, I can backport it from a 4.1.x PR milestoned back to 4.0.x, but we need to remember about it as I recall the scripts cannot handle that case

Reply:
So... This bug was introduced in #10797 (milestoned for 4.0.4) but it is not yet in the v4.0.x branch. What do I do now?

Reply:
Anyways, see #10945